lib/render/pxrUsdMayaGL/batchRenderer.cpp

```diff
@@ -1129,6 +1136,27 @@ UsdMayaGLBatchRenderer::_TestIntersection(
                  GL_TEXTURE_BIT |
                  GL_POLYGON_BIT);
 
+    // XXX: When Maya is using OpenGL Core Profile as the rendering engine (in
+    // either compatibility or strict mode), batch renders like those done in
+    // the "Render View" window or through the ogsRender command do not
+    // properly track uniform buffer binding state. This was causing issues
+    // where the first batch render performed would look correct, but then all
+    // subsequent renders done in that Maya session would be completely black
+    // (no alpha), even if the frame contained only Maya-native geometry or if
+    // a new scene was created/opened.
+    //
+    // To avoid this problem, we need to save and restore Maya's bindings
+    // across Hydra calls. We try not to bog down performance by saving and
+    // restoring *all* GL_MAX_UNIFORM_BUFFER_BINDINGS possible bindings, so
+    // instead we only do just enough to avoid issues. Empirically, the
+    // problematic binding has been the material binding at index 4.
```

Comment:
Looks like this would only affect selection performance? Any other workflow that would be impacted?

Reply:
The one other workflow that might be affected is when live surface is used. We end up using Hydra to do the intersection testing for that as well:
https://github.com/Autodesk/maya-usd/blob/c75e69638c560a82163f449cbcc3cbfbd8077e27/lib/render/pxrUsdMayaGL/proxyShapeDelegate.cpp#L96